docs/companion.md

```diff
@@ -3,3 +3,683 @@ sidebar_position: 4
+
+## Installation
+
+If you want to host your own Companion server start by installing it:
+
+```bash
+npm install @uppy/companion
+```
+
+:::note
+Since v2, you need to be running `node.js >= v10.20.1` to use Companion.
+More information in the [migrating to 2.0](/docs/guides/migrate-2.0) guide.
+
+Windows is not a supported platform right now.
+It may work, and we’re happy to accept improvements in this area, but we can’t provide support.
+:::
+
+## Use
+
+Companion can be integrated as middleware into your [Express](https://expressjs.com/) app or as a standalone server.
+
+### Express middleware
+
+To plug Companion into an existing server, call its `.app` method,
+passing in an [options](#Options) object as a parameter.
+This returns a server instance that you can mount on a route in your Express app.
+
+```js
+import express from 'express'
+import bodyParser from 'body-parser'
+import session from 'express-session'
+import companion from '@uppy/companion'
+
+const app = express()
+
+// Companion requires body-parser and express-session middleware.
+// You can add it like this if you use those throughout your app.
+//
+// If you are using something else in your app, you can add these
+// middlewares in the same subpath as Companion instead.
+app.use(bodyParser.json())
+app.use(session({ secret: 'some secrety secret' }))
+
+const options = {
+  providerOptions: {
+    drive: {
+      key: 'GOOGLE_DRIVE_KEY',
+      secret: 'GOOGLE_DRIVE_SECRET',
+    },
+  },
+  server: {
+    host: 'localhost:3020',
+    protocol: 'http',
+    // This MUST match the path you specify in `app.use()` below:
+    path: '/companion',
+  },
+  filePath: '/path/to/folder/',
+}
+
+app.use('/companion', companion.app(options))
+```
+
+Companion uses WebSockets to communicate progress, errors, and successes to the client.
+This is what Uppy listens to to update it’s internal state and UI, if there is one.
+
+Add the Companion WebSocket server using the `companion.socket` function:
+
+```js
+const server = app.listen(PORT)
+
+companion.socket(server)
+```
+
+If WebSockets fail for some reason Uppy and Companion will fallback to HTTP polling.
+
+### Standalone
+
+You can use the standalone version if you want to run Companion as it’s own Node process.
+The standalone version is a configured Express server with sessions, logging, and security best practices.
+
+Companion ships with an excecutable file (`bin/companion`) which is the standalone server.
+Unlike the middleware version, options are set via environment variables.
+
+:::info
+Checkout [options](#options) for the available options in JS and environment variable formats.
+:::
+
+You need at least these three to get started:
+
+```bash
+export COMPANION_SECRET="shh!Issa Secret!"
+export COMPANION_DOMAIN="YOUR SERVER DOMAIN"
+export COMPANION_DATADIR="PATH/TO/DOWNLOAD/DIRECTORY"
+```
+
+Then run:
+
+```bash
+companion
+```
+
+You can also pass in the path to your JSON config file, like so:
+
```

Comment:
```suggestion

:::info
Companion will run as http and expects a reverse proxy with SSL termination in front of it when running in production. See `COMPANION_PROTOCOL` [option](#options) for more information.
:::
```

because someone was confused by this here: https://community.transloadit.com/t/companion-generates-redirect-urls-without-ssl/16264

Reply:
I moved this up, see my other review comment